### PR TITLE
Terraform: Storage Pool Resource

### PR DESCRIPTION
**Pull request by vihangarao1**
# Description

- It includes Storage Pool Modify Name function
- Modify Media Type
- Set Read Ram Cache
- Enable/Disable RF Cache.
- Added Integrated test cases for the API.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Test A
- [x] Test B

### CODE REVIEW

File: inttests/GOSCALEIO_TEST.env [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Make sure system has a name (not just an ID) for tests to pass
+GOSCALEIO_INSECURE=true
+GOSCALEIO_HTTP_TIMEOUT=60
+GOSCALEIO_ENDPOINT=https://10.60.156.69/
```
Comment (JacobGros) [inline inttests/GOSCALEIO_TEST.env:4]: we shouldn't upload internal IPs / specifics to public repos

File: inttests/GOSCALEIO_TEST.env_example [hunk 2]
```diff
@@ -1,9 +1,9 @@
 # Make sure system has a name (not just an ID) for tests to pass
 GOSCALEIO_INSECURE=true
```
Comment (JacobGros) [inline inttests/GOSCALEIO_TEST.env_example:2]: We should keep the env example file generic

File: inttests/GOSCALEIO_TEST.env (hunk 1 above)
Comment (vihangarao1) [reply inttests/GOSCALEIO_TEST.env]: Updated

**Review by khareRajshree — CHANGES_REQUESTED** [review]
Please add a description about the changes done for this PR, also checklist and how the changes were tested. Also refrain using custom names for PR names.
Thanks.

File: storagepool.go [hunk 3]
```diff
@@ -66,6 +66,96 @@ func (pd *ProtectionDomain) CreateStoragePool(name string, mediaType string) (st
...
+
+// Modify storagepool RMcache
+func (sp *StoragePool) ModifyRMCache(useRmcache string) error {
+
+	link, err := GetLink(sp.StoragePool.Links, "self")
+	if err != nil {
+		return err
+	}
+	path := fmt.Sprintf("%v/action/setUseRmcache", link.HREF)
+	fmt.Println(path)
```
Comment (VamsiSiddu-7) [inline storagepool.go:115]: remove the debug statement

File: storagepool.go [hunk 4]
```diff
@@ -66,6 +66,96 @@ func (pd *ProtectionDomain) CreateStoragePool(name string, mediaType string) (st
...
+	payload := &types.StoragePoolUseRmCache{
+		UseRmcache: useRmcache,
+	}
+	err1 := sp.client.getJSONWithRetry(
```
Comment (VamsiSiddu-7) [inline storagepool.go:119]: why do we need err1 can't we reuse the var err defined on  line 110.

File: types/v1/types.go [hunk 5]
```diff
@@ -395,6 +395,21 @@ type StoragePoolParam struct {
...
+
+type StoragePoolMediaType struct {
+	MediaType string `json:"mediaType"`
+}
+
+type StoragePoolUseRmCache struct {
+	UseRmcache string `json:"useRmcache"`
+}
+
+type StoragePoolUseRfCache struct {
```
Comment (VamsiSiddu-7) [inline types/v1/types.go:410]: is it supposed to be an empty struct?

File: inttests/storagepool_test.go [hunk 6]
```diff
@@ -223,3 +223,59 @@ func TestCreateDeleteStoragePool(t *testing.T) {
 	assert.NotNil(t, err)
 
 }
+
```
Comment (khareRajshree) [inline inttests/storagepool_test.go:226]: Please remove the print statements if they are for debugging purpose.

**Review by khareRajshree — CHANGES_REQUESTED** [review]

**Review by VamsiSiddu-7 — CHANGES_REQUESTED** [review]
please address the review comments.

**Review by khareRajshree — COMMENTED** [review]
Please add signed/verified commits.

File: types/v1/types.go (hunk 5 above)
Comment (vihangarao1) [reply types/v1/types.go:410]: Yes, it has to be an empty struct

Comment (VamsiSiddu-7) [reply types/v1/types.go:410]: okay

**Comment by VamsiSiddu-7** [comment]
you dont need to raise pr from the fork. you can directly create the branch in the actual repo an raise a pr.

**Comment by vihangarao1** [comment]
Closing pull request and raising new request

**Pull request closed without merging**